### PR TITLE
- ec2uploadimg

### PR DESCRIPTION
**Pull request by rjschwei**
+ Support uploading with helper instances that handle disk attachments
    via NVMe protocol
  + Improve output of help for clarity
  + Improve progress message for ssh connection to make it easier to
    discover issues
  + Fix implementation for using root-swap method with a running instance

### CODE REVIEW

File: lib/ec2imgutils/ec2uploadimg.py [hunk 1]
```diff
@@ -568,7 +577,7 @@ def _establish_ssh_connection(self):
         client = paramiko.client.SSHClient()
         client.set_missing_host_key_policy(paramiko.WarningPolicy())
         if self.verbose:
-            print('Attempt ssh connection')
+            print('Attempt ssh connection to %s' % instance_ip)
```
Comment (smarlowucf) [inline lib/ec2imgutils/ec2uploadimg.py]: Suggest moving towards format string vs %s.

File: lib/ec2imgutils/ec2uploadimg.py [hunk 2]
```diff
@@ -621,19 +630,27 @@ def _execute_ssh_command(self, command):
...
-        if self.aborted:
-            return
-        device_letter = device_id[-1]
-        expected_name = '/dev/xvd%s' % device_letter
-        if self._device_exists(expected_name):
-            return expected_name
+    def _find_device_name(self, device_size):
+        """Match an attached volume by size"""
+        lsblk_out = json.loads(self._execute_ssh_command('lsblk -a -J'))
+        for device in lsblk_out['blockdevices']:
```
Comment (smarlowucf) [inline lib/ec2imgutils/ec2uploadimg.py]: Should this be checked first that there is a list of blockdevices? Or is it guaranteed to be an empty list if none are returned?

File: lib/ec2imgutils/ec2uploadimg.py [hunk 3]
```diff
@@ -92,6 +93,8 @@ def __init__(self,
         self.default_sleep = 10
         self.device_ids = ['f', 'g', 'h', 'i', 'j']
         self.instance_ids = []
+        if self.running_id:
```
[the remark this answers is not in the document]
Comment (rjschwei) [reply lib/ec2imgutils/ec2uploadimg.py]: @smarlowucf do you think it is worth changing the interface of the calss for clarification, i.e. major version bump?

[the remark this answers is not in the document]
Comment (smarlowucf) [reply lib/ec2imgutils/ec2uploadimg.py]: Not if the name already existed.

**Pull request merged**